discovery-provider/src/utils/helpers.py

```diff
@@ -247,7 +247,9 @@ def multihash_digest_to_cid(multihash_digest):
 
 
 def get_web3_endpoint(shared_config):
-    if shared_config["web3"]["port"] != '443':
+    if shared_config["web3"]["provider_url"]:
+        web3endpoint = shared_config["web3"]["provider_url"]
+    elif shared_config["web3"]["port"] != '443':
```

Comment:
was this change required?
/ can you document

Reply:
This is so that the code is backward compatible with the old configs.

Added a small comment to explain this 👍